src/gtk/toga_gtk/widgets/canvas.py

```diff
@@ -22,9 +26,29 @@ def gtk_draw_callback(self, canvas, gtk_context):
         using the canvas and gtk_context function arguments. This method calls
         the draw method on the interface Canvas to draw the objects.
 
+        Sine Gtk+ does not include a dedicate event for `resize` the way to
+        emulate it is by checking if the widget size was changed while
+        redrawing it. This works because resizing a widget in GTK also results
+        in a draw event.
```

Comment:
Are you sure about this? The [GtkDrawingArea docs](https://developer.gnome.org/gtk3/stable/GtkDrawingArea.html) suggest there is a `size-allocate` signal to catch when the widget changes size...

Reply:
hmm, yeah that seems to work, not sure why I thought it wouldn't when I looked into that before. Fixed.